### ISSUE TITLE
Research Monero

### ISSUE THREAD

**Issue by mitexleo** [horizontalsystems/unstoppable-wallet-android#5492]
Hi 👋. 
Why UW doesn't support 😕 Monero , Oxen and LBC ?
Is it possible to add support for these currencies?
![SmartSelect_20221106-134859_Unstoppable](https://user-images.githubusercontent.com/114842102/200159967-ca93208a-325f-4022-af2a-2396b1d590bd.jpg)

**Comment by esengulov** [comment]
Monero support is on the roadmap. We need more time

**Comment by neonota** [comment]
> Monero support is on the roadmap. We need more time

Thanks 😊 🫂...

**Comment by TheFuzzStone** [comment]
> We need more time

Do you have at least a rough timeline for when Monero (XMR) support will appear?

Again, all the other mobile multicurrency wallets that support XMR are all proprietary, and Monero community has had problems with all of them. From last problems - developers of Exodus, Coinomi, Trustee hardcoded their Monero nodes, but didn't update them (though they were warned by Monero developers and community) and many users faced with inability to use their XMR.

So your advice - make it possible for Monero users to choose which node to connect to themselves, as it is done in mobile wallets Monerujo and Cake Wallet.

**Comment by TheFuzzStone** [comment]
> Monero support is on the roadmap. We need more time

Do you have at least a rough timeline for when Monero (XMR) support will appear?

All the other mobile multicurrency wallets that support XMR are all proprietary, and Monero community has had problems with all of them. From last problems - developers of Exodus, Coinomi, Trustee hardcoded their Monero nodes, but didn't update them (though they were warned by Monero developers and community) and many users faced with inability to use their XMR.

So your advice - make it possible for Monero users to choose which node to connect to themselves, as it is done in mobile wallets Monerujo and Cake Wallet.

**Comment by fork995** [comment]
I'm looking forward to monero integration. That's a great idea👍🏻

**Comment by ORGaniZMo** [comment]
c'mon, devs! Monero is very much needed in your system!

**Comment by sandbird75** [comment]
Where is Monero? We want to use it.

**Comment by multed** [comment]
I, like many, am still waiting for Monero support for my wallet.

**Comment by esengulov** [comment]
> Hey @esengulov what is the status of your XMR studies? Could you give us some news, leaks about it? Will it does? 😁

Unfirtunatelky we werent able to start researching it yet. As soon as we are done with TON which is nearing completion the next blockchain on the line is Monero

**Comment by candroid-man** [comment]
Any updates yet?

**Comment by oblak-be** [comment]
Also looking forward for Monero support!!!

**Comment by TheFuzzStone** [comment]
@esengulov, any updates?

**Comment by notpri** [comment]
> I have definitively leaved the Monero project. Like ZEC, never evoluted in something really useful.
> 
>     1. Transactions are incredibly slow and expensive (not ideal solution for on street purchases)
> 
>     2. There are a lot of problems to exchange it for normal user
> 
>     3. Balance not immediately available after that a previous transaction is sent recently
> 
>     4. PoW is an energy disaster movie (also $KAS has this issue)
> 
> 
> There are a lot of new currencies that have potential (INJ, SEI etc.), and $ZEC will have my attention again when it turns its consensus mechanism to DPoS or something similar.
> 
> These are my ideas. Start with thumbs down.

I also don't hold monero now, I hold Kaspa (KAS). INJ is also great. I would love to see Kaspa first and INJ second in UW.

**Comment by kdejaeger** [comment]
Those are just tradeoffs for ultimate privacy I guess ... .

**Comment by OrginalS** [comment]
@serrq  I don't mean to turn this issue into a discussion board about monero, but almost everything you said is wrong.

> I have definitively leaved the Monero project. Like ZEC, never evoluted in something really useful.

Monero is one of the most used coins when it comes to actual payments. According to business owners like Luke Smith, coincards or GrapheneOS most of their transactions/donations are done with monero [1](https://twitter.com/CoinCards/status/1508324025256202246) [2](https://twitter.com/CoinCards/status/1626105845003210752) [3](https://twitter.com/CoinCards/status/1736483739285250056) [4](https://twitter.com/CoinCards/status/1521549401860190208) [5](https://twitter.com/GrapheneOS/status/1684663777273233408).

> 1. Transactions are incredibly slow and expensive (not ideal solution for on street purchases)

Monero aims for 2 minute block times, so on average it takes 1 minute till 1 conf, but because monero doesn't support RBF, you can safely accept a 0 conf transaction for small amounts. An average monero transaction fee is about $0.05 usd right now.

> 2. There are a lot of problems to exchange it for normal user

If you only ever use 1 cex to do all your crypto business, then sure, it can be annoying for new users to buy monero, but it really isn't as difficult as some people make it out to be. You can easily exchange btc, ltc, eth, usdt... on almost any dex or cex, like tradeogre, nonkyc.io, etc., via atomic swaps or buy it directly P2P on sites like local monero and soon haveno and bisq.

> 3. Balance not immediately available after that a previous transaction is sent recently

There is a 10 block long lock on new funds for privacy purposes, but all the funds you already have in your wallet are fully available. Because Monero aims for 2 minute blocks, it will take an average of 9 minutes for your new funds to unlocks, which is less than the average time for 1 bitcoin block.

> 4. PoW is an energy disaster movie (also $KAS has this issue)

There are pros and cons to PoW but when it comes to e-waste or energy consumption monero is one of the least "wasteful" coins. It is mined with consumer grade cpus instead of gpus or asics so it uses way less energy for mining and no resources for asic hardware production that will be obsolete in 1-2 years. Because you can't really use multiple cpus on a single motherboard, it doesn't make sense to build a mining farm specifically for Monero. This means that it's primarily mined by regular people with their home PCs that they already own, which is great for the environment and decentralization.

**Comment by TheFuzzStone** [comment]
@esengulov, @abdrasulov, @Dianaaiym, any updates?

**Comment by TheFuzzStone** [comment]
So, I understand right, that your team has chosen the tactic of not responding?

@esengulov, @abdrasulov, @Dianaaiym

**Comment by esengulov** [comment]
> So, I understand right, that your team has chosen the tactic of not responding?
> 
> @esengulov, @abdrasulov, @Dianaaiym

my bad, had no intention to ignore! We are not that good with monitoring github and repsonding in timely manner. Best would be to ping on telegram as we are ratrher prompt there. 

Let me assure that Monero is on the priority list when it comes to next blockchains to add. The moment we get 3-4 weeks spare slot we are going to put someone on it and implement it. Hopefully this will be done during summer months.

**Comment by TheFuzzStone** [comment]
> > So, I understand right, that your team has chosen the tactic of not responding?
> > @esengulov, @abdrasulov, @Dianaaiym
> 
> my bad, had no intention to ignore! We are not that good with monitoring github and repsonding in timely manner. Best would be to ping on telegram as we are ratrher prompt there.
> 
> Let me assure that Monero is on the priority list when it comes to next blockchains to add. The moment we get 3-4 weeks spare slot we are going to put someone on it and implement it. Hopefully this will be done during summer months.

I realize that you have a lot of work to do and that it can be hard to keep track of everything.

Thank you for your reply. This is very cool news and I will do my best to make your awesome wallet known to even more people.

A very important thing is that in “Blockchain settings” the user should be able to manually add/change Monero node. This is important because all the responsibility will be on the user and not on you as the wallet provider. This stupidity was done by other proprietary wallets, and then when they forgot to update their hard-coded nodes - users just couldn't use their funds.

Anyway, I will be the first user to test how Monero works in your wallet, and if I have any suggestions for improvements and/or bugs - I will definitely help you with them. 

Thanks again.

**Comment by TheFuzzStone** [comment]
![image](https://github.com/user-attachments/assets/1249a55e-13c2-4061-8fd9-510be1ab6942)

Please decrypt. So, Monero to wait at 0.41 or not to wait?

**Comment by kerlann777** [comment]
Monero please, waiting for...

**Comment by BDawson89** [comment]
Yes please add Monero!! I appreciate all you and your team does btw!
